### PR TITLE
[systemtest] Refactor metrics tests

### LINKED ISSUE #7544: Fix and re-enable `MetricsIsolatedST.testKafkaBridgeMetrics` test
The `MetricsIsolatedST.testKafkaBridgeMetrics` is disabled with https://github.com/strimzi/strimzi-kafka-operator/pull/7543.
After digging into it and fixing the issue we should re-enable the test again.

### PR DESCRIPTION
**Pull request by im-konge**
Signed-off-by: Lukas Kral <lukywill16@gmail.com>

### Type of change

- Bugfix
- Refactoring

### Description

This PR completely refactors the whole `MetricsIsolatedST` suite, which is flaky. As part of this refactoring, I added `collectedData` into the `MetricsCollector`, so each instance holds latest version of metrics get from each `resource` or `pod`. When the `collectedData` doesn't contain specific pattern, the `waitFor` is used here for specific metric to appear. For each poll, `collectedData` is updated with new value. This way I removed flakiness in case the metric wasn't present just in the moment of assert.

Also, this PR removes some checks, which weren't working before - for example checking metrics of resources, which doesn't exist.

As part of this PR, I removed few metrics, which were not relevant (because it depended or environment or resource and it's state):

`system_cpu_count`
`strimzi_reconciliations_locked_total`
`strimzi_reconciliations_failed_total` - if reconciliation doesn't fail, this metric is not present in metrics
`strimzi_reconciliations_periodical_total` for KafkaConnector, which is not present

Fixes #7544 

### Checklist

- [ ] Make sure all tests pass

### CODE REVIEW

**Review by Frawless — APPROVED** [review]

File: systemtest/src/main/java/io/strimzi/systemtest/metrics/MetricsCollector.java [hunk 1]
```diff
@@ -208,21 +215,51 @@ private int getDefaultMetricsPortForComponent() {
...
     }
 
+    /**
+     * Method checks already collected metrics data for Pattern containing desired metric
+     * @param pattern Pattern of metric which is desired
+     *
+     * @return ArrayList of values collected from the metrics
+     */
+    public ArrayList<Double> waitForSpecificMetricAndCollect(Pattern pattern) {
+        synchronized (LOCK) {
```
Comment (see-quick) [inline systemtest/src/main/java/io/strimzi/systemtest/metrics/MetricsCollector.java:242]: You can use synchronized keyword in the defining method scope:
```java
public ArrayList<Double> waitForSpecificMetricAndCollect(Pattern pattern) 
```
--
```java
public synchronized ArrayList<Double> waitForSpecificMetricAndCollect(Pattern pattern) 
```

**Review by see-quick — APPROVED** [review]
Great work! 🕺👍 One nit...

File: systemtest/src/main/java/io/strimzi/systemtest/metrics/MetricsCollector.java (hunk 1 above)
Comment (im-konge) [reply systemtest/src/main/java/io/strimzi/systemtest/metrics/MetricsCollector.java:242]: Good point, thanks! :)

**Pull request merged**